server/src/main/java/com/paypal/selion/proxy/SeLionRemoteProxy.java

```diff
@@ -224,37 +247,97 @@ public synchronized void shutdownNode() {
         url.append("/extra/");
         url.append(NodeForceRestartServlet.class.getSimpleName());
         HttpPost post = new HttpPost(url.toString());
+        int responseStatusCode = HttpStatus.SC_NOT_FOUND;
         try {
-            client.execute(post);
-            appendMsgToCustomLog("Node " + machine + " shut-down successfully.");
+            HttpResponse response = client.execute(post);
+            responseStatusCode = response.getStatusLine().getStatusCode();
         } catch (IOException e) {
             LOGGER.log(Level.SEVERE, e.getMessage(), e);
         } finally {
+            if (responseStatusCode == HttpStatus.SC_OK) {
+                proxyLogger.info("Node " + machine + " shutdown successfully.");
+            } else {
+                proxyLogger.info("Node " + machine + " did not shutdown. Return code was " + responseStatusCode);
+            }
+
             try {
                 client.close();
             } catch (IOException e) {
                 LOGGER.log(Level.SEVERE, e.getMessage(), e);
             }
         }
+        stopNodeRecycleThread();
+        // remove the node from the hub registry, if it reported 200 OK for NodeForceRestartServlet request
+        if (responseStatusCode == HttpStatus.SC_OK) {
+            getRegistry().removeIfPresent(this);
```

Comment:
Already do. `requestNodeShutdown` (graceful shutdown) starts the `NodeRecycleThread` which calls `forceNodeShutdown` upon exit and when not interrupted.